config.default

```diff
@@ -132,6 +132,15 @@ mpasInterpolationMethod = bilinear
 startYear = 1
 endYear = 9999
 
+[index]
+## options related to producing nino index.
+
+# start and end years for the nino 3.4 analysis.  Using out-of-bounds values
+#   like start_year = 1 and end_year = 9999 will be clipped to the valid range
+#   of years, and is a good way of insuring that all values are used.
+startYear = 2
+endYear = 6
```

Comment:
The idea is to keep everything in config.defaults, but then re-define the options that the user is likely to have to modify for the specific run in the configs/$machine examples. Things like file locations, of course, but also start and end year for the various sections, directory paths, etc.
This is something that @xylar introduced recently and it turns out to be a time saver because before we had to update all the config examples, even when making changes to the config file that the user was unlikely to have to change routinely.